spec/factories/users.rb

```diff
@@ -0,0 +1,48 @@
+require 'faker'
+
+FactoryBot.define do
+  factory :user do
+    name { Faker::Name.unique.name }
+    photo { Faker::Avatar.image }
+    bio { Faker::Lorem.sentence }
+
+    factory :user_with_posts do
+      transient do
+        post_count { 3 }
+        comments_count { 5 }
+      end
+
+      after(:create) do |user, evaluator|
+        create_list(:post_with_comments, evaluator.post_count, author: user, comments_count: evaluator.comments_count)
+      end
+    end
+  end
+end
+
+FactoryBot.define do
+  factory :post do
+    title { 'hello there' }
+    text { 'this is my first post' }
+    comments_counter { 0 }
+    likes_counter { 0 }
+    association :author, factory: :user
+
+    factory :post_with_comments do
+      transient do
+        comments_count { 5 }
+      end
+
+      after(:create) do |post, evaluator|
+        create_list(:comment, evaluator.comments_count, post:)
```

Comment:
- [ Optional]  for the `post:` has no value is causing the error while running the app. Please fix it so that the app can run properly without issues.

Reply:
> * [ Optional]  for the `post:` has no value is causing the error while running the app. Please fix it so that the app can run properly without issues.

@ishimwezachee, this is a bit confusing, it is actually rubucop that tells me that assignment is useless and suggests I remove, on mind end the test all pass even if don't assign any value to post